### ISSUE TITLE
self-host picture storage

### ISSUE THREAD

**Issue by heylix** [flint-gg/easyshare#17]
You could give any member ~50MB of free storage so downloads can happen directly from your site.

This would also increase engagement with your site and the possibility to push more features to the users.

**Comment by T0TProduction** [comment]
Sounds awesome, but offering storage is a huge hosting cost.
Another issue is data security etc. since we're from Germany. If we don't store it ourselves, we don't need to worry about so many things.

For now, we won't be able to self-host the media, but if we somehow get funding this definitely is something i would want to add support for. 
Maybe even a premium-user thing, where if you don't want to use third-party apps like Google Photos, for a small monthly or yearly fee, we host your media. (idk what this would be, maybe 1€ a month?)

**Comment by heylix** [comment]
>If we don't store it ourselves, we don't need to worry about so many things.

You're kind of using a third party to forward potentially PII to, so you might have to have an Auftragsdatenverarbeitungsvertrag with Google anyway. You may want to check this with a lawyer, IANAL.

>offering storage is a huge hosting cost.

50MB per user and a maximum age of, say, 1 week should be enough, IMHO. Another idea would be that your FOSS components allow self-hosting the media. Either way, I don't want to have to use Google for this.

Arbitrary endpoints like (S)FTP, HTTP POST, [linx-server](https://github.com/andreimarcu/linx-server) or Nextcloud/WebDAV would be another idea. I could've named the issue "support something else than Google" but IMHO the easiest implementation would be mail (#18) or self-hosting.

Thank you for considering.

**Comment by T0TProduction** [comment]
> You're kind of using a third party to forward potentially PII to, so you might have to have an Auftragsdatenverarbeitungsvertrag with Google anyway. You may want to check this with a lawyer, IANAL.

Good point, the legal issue is something that can always be problematic, and I think for now we will just work in the best interest of the user, hoping that is enough.

Concerning the other point:
You're right, and supporting **something** besides Google Photos is definitely going to happen.
Maybe adding this hosting with the limits you mentioned can be done cost-efficient enough. 
The solution of adding support for self-hosting sounds like an even greater idea, but I personally don't have experience setting these kinds of systems up, and I wonder how many of the users are actually able and/or willing to set something up. So yes, I would want to support it, but maybe the priority is really low.

The solution I had in mind up until now was to simply add support for other cloud services ( e.g. Apple Photos, Dropbox, etc.. )
IMO that's the easiest thing for users, as pretty much everyone has at least one cloud service they already use, and connecting APIs seems a lot easier than both
- hosting images ourselves and managing access, deletion, etc.
- enabling self-hosting

But, if e.g. you have experience with self-hosting, maybe you could take on that topic and together we can build a system that allows users to add their own cloud.
